pkg/counter/counter.go

```diff
@@ -12,73 +12,74 @@ type Counter struct {
 	value uint64
 }
 
-// NewCounter creates a new counter with the given initial value.
 func NewCounter(initialValue uint64) Counter {
 	return Counter{value: initialValue}
 }
 
-// Increase increases the counter by the given value (thread-safe).
-func (c *Counter) Increase(values ...uint64) error {
-	_, err := c.IncreaseAndRead(values...)
-	return err
-}
+// Increment
 
-// IncreaseAndRead increases the counter by the given value (thread-safe) and returns the new value.
-func (c *Counter) IncreaseAndRead(values ...uint64) (uint64, error) {
-	if len(values) == 0 {
-		values = append(values, 1)
+// Increment increments the counter by given value (default: 1, thread-safe).
+func (c *Counter) Increment(x ...uint64) error {
+	val := uint64(1)
+	if len(x) != 0 {
+		for _, v := range x {
+			val += v
+		}
+		val--
 	}
+	_, err := c.IncrementValueAndRead(val)
+	return err
+}
 
-	var n uint64
+// IncrementValueAndRead increments the counter by given value and returns the new value (thread-safe).
+func (c *Counter) IncrementValueAndRead(x uint64) (uint64, error) {
```

Comment:
Here we can simplify to:

```go
func (c *Counter) IncrementValueAndRead(x uint64) (uint64, error) {
	n := atomic.AddUint64(&c.value, x)
	if n < x {
		return n, errors.New("counter overflow")
	}

	return n, nil
}
```

Reply:
The only logical difference I see, despite styling, is not checking x > 0 (which is actually a leftover from other attempts of mine). If that is the case, than yep, that can be removed.

Reply:
Yep, the suggestion was merely figurative as I copied n pasted local changes. Ignore the styling changes.

Reply:
Okay then, removing the x > 0 leftover, thx.